apps/admin/frontend/src/screens/contest_adjudication_screen.tsx

```diff
@@ -0,0 +1,1124 @@
+    writeIns,
+    writeInCandidates,
+  ]);
+
+  // Initiate cvr scrolling
+  useEffect(() => {
+    if (
+      writeInCvrQueueQuery.isSuccess &&
+      firstPendingWriteInCvrIdQuery.isSuccess &&
+      !scrollStateInitialized
+    ) {
+      const cvrQueue = writeInCvrQueueQuery.data;
+      const firstPendingWriteInCvrId = firstPendingWriteInCvrIdQuery.data;
+      if (firstPendingWriteInCvrId) {
+        setScrollIndex(cvrQueue.indexOf(firstPendingWriteInCvrId));
+      } else {
+        setScrollIndex(0);
+      }
+    }
+  }, [
+    firstPendingWriteInCvrIdQuery,
+    scrollStateInitialized,
+    writeInCvrQueueQuery,
+  ]);
+
+  // Scroll candidate list to write-ins if adjudications are required
+  const candidateListRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    if (shouldAutoscrollUser && candidateListRef.current) {
+      candidateListRef.current.scrollTop =
+        candidateListRef.current.scrollHeight;
+      setShouldAutoscrollUser(false);
+    }
+  }, [shouldAutoscrollUser]);
+
+  // Prefetch the next and previous ballot images
+  const queryClient = useQueryClient();
+  const apiClient = useApiClient();
+  useEffect(() => {
+    if (!writeInCvrQueueQuery.isSuccess || !scrollStateInitialized) return;
+    function prefetch(cvrId: Id) {
+      void queryClient.prefetchQuery({
+        queryKey: getCvrWriteInImageViews.queryKey({ cvrId, contestId }),
+        queryFn: () =>
+          apiClient.getCvrContestWriteInImageViews({ cvrId, contestId }),
+      });
+    }
+    const nextCvrId = writeInCvrQueueQuery.data[scrollIndex + 1];
+    if (nextCvrId) {
+      prefetch(nextCvrId);
+    }
+    const prevCvrId = writeInCvrQueueQuery.data[scrollIndex - 1];
+    if (prevCvrId) {
+      prefetch(prevCvrId);
+    }
+  }, [
+    apiClient,
+    contestId,
+    queryClient,
+    scrollIndex,
+    scrollStateInitialized,
+    writeInCvrQueueQuery,
+  ]);
+
+  const areQueriesLoading =
+    !firstPendingWriteInCvrIdQuery.isSuccess ||
+    !writeInCvrQueueQuery.isSuccess ||
+    !cvrVoteInfoQuery.isSuccess ||
+    !cvrContestWriteInImagesQuery.isSuccess ||
+    !cvrContestWriteInsQuery.isSuccess ||
+    !contestWriteInCandidatesQuery.isSuccess ||
+    !cvrContestVoteAdjudicationsQuery.isSuccess;
+
+  if (
+    !(scrollStateInitialized && voteStateInitialized) ||
+    (areQueriesLoading && !isStateStale)
+  ) {
+    return (
+      <NavigationScreen title="Contest Adjudication">
+        <Loading isFullscreen />
+      </NavigationScreen>
+    );
+  }
+
+  return (
+    <Screen>
+      <Main flexRow>
+        <BallotPanel>
+          {isFocusedWriteInHmpb && (
+            <BallotZoomImageViewer
+              ballotBounds={firstWriteInImage.ballotCoordinates}
+              key={currentCvrId} // Reset zoom state for each write-in
+              imageUrl={firstWriteInImage.imageUrl}
+              zoomedInBounds={
+                focusedWriteInImage &&
+                'writeInCoordinates' in focusedWriteInImage
```

Comment:
Consider adding a `type` property to the `WriteInImageView` types so we don't have to property snoop here.